src/Common/src/Interop/Unix/System.Native/Interop.readdir.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal enum NodeType : short
+        {
+            Unknown         = 0,
+            NamedPipe       = 1,
+            CharacterDevice = 2,
+            Directory       = 4,
+            BlockDevice     = 6,
+            File            = 8,
+            SymLink         = 10,
+            Socket          = 12,
+            BsdWhiteout     = 14,
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct DirectoryEntry
+        {
+            internal NodeType   inode_type;
+            internal IntPtr     nameStartAddress;
+            internal int        nameLength;
+        }
```

Comment:
I've been using PascalCase for struct members on both C# and C++ side.

I'll write up some more detailed guidelines soon.